src/main/java/slimeknights/tconstruct/smeltery/tileentity/TileSearedFurnace.java

```diff
@@ -0,0 +1,562 @@
+   */
+  public float getSmeltingProgress(int index) {
+    if(index < 0 || index > getSizeInventory() - 1) {
+      return -1f;
+    }
+
+    if(!canHeat(index)) {
+      return -1f;
+    }
+
+    return getProgress(index);
+  }
+  
+  @Override
+  public int getTemperature(int i) {
+    if(i < 0 || i >= itemTemperatures.length) {
+      return 0;
+    }
+
+    return itemTemperatures[i];
+  }
+
+  public int getTempRequired(int i) {
+    if(i < 0 || i >= itemTempRequired.length) {
+      return 0;
+    }
+    return itemTempRequired[i];
+  }
+
+  @SideOnly(Side.CLIENT)
+  public FuelInfo getFuelDisplay() {
+    FuelInfo info = new FuelInfo();
+
+    // we still have leftover fuel
+    if(hasFuel()) {
+      info.fluid = currentFuel.copy();
+      info.fluid.amount = 0;
+      info.heat = this.temperature;
+      info.maxCap = currentFuel.amount;
+    }
+    else if(currentTank != null) {
+      // we need to consume fuel, check the current tank
+      if(hasFuel(currentTank, currentFuel)) {
+        IFluidTank tank = getTankAt(currentTank);
+        info.fluid = tank.getFluid().copy();
+        info.heat = temperature;
+        info.maxCap = tank.getCapacity();
+      }
+    }
+
+    // check all other tanks (except the current one that we already checked) for more fuel
+    for(BlockPos pos : tanks) {
+      if(pos == currentTank) {
+        continue;
+      }
+
+      IFluidTank tank = getTankAt(pos);
+      // tank exists and has something in it
+      if(tank != null && tank.getFluidAmount() > 0) {
+        // we don't have fuel yet, use this
+        if(info.fluid == null) {
+          info.fluid = tank.getFluid().copy();
+          info.heat = info.fluid.getFluid().getTemperature(info.fluid);
+          info.maxCap = tank.getCapacity();
+        }
+        // otherwise add the same together
+        else if(tank.getFluid().isFluidEqual(info.fluid)) {
+          info.fluid.amount += tank.getFluidAmount();
+          info.maxCap += tank.getCapacity();
+        }
+      }
+    }
+
+    return info;
+  }
+  
+  /* Networking and saving */
+  @SideOnly(Side.CLIENT)
+  public void updateFuelFromPacket(int index, int fuel) {
+    if(index == 0) {
+      this.fuel = fuel;
+    }
+    else if(index == 1) {
+      this.fuelQuality = fuel;
+    }
+  }
+  
+  @SideOnly(Side.CLIENT)
+  public void updateTemperatureFromPacket(int index, int heat) {
+    if(index < 0 || index > getSizeInventory() - 1) {
+      return;
+    }
+  
+    itemTemperatures[index] = heat;
+  }
+  
+  @SideOnly(Side.CLIENT)
+  public void updateTempRequiredFromPacket(int index, int heat) {
+    if(index < 0 || index > getSizeInventory() - 1) {
+      return;
+    }
+  
+    itemTempRequired[index] = heat;
+  }
+  
+  @Override
+  public void validate() {
+    super.validate();
+    // on validation we set active to false so the furnace checks anew if it's formed
+    active = false;
+  }
+  
+  public static final String TAG_FUELQUALITY = "fuelQuality";
```

Comment:
:+1: very good. Move to top of class or rather to the intermediate layer mentioned above since all that stuff is needed for everything anyway. Also `smelted` and `tankSide` are unused